third_party/pjs/promise.js

```diff
@@ -0,0 +1,420 @@
+/**
+ * Constructs a ES6/Promises A+ Promise instance.
+ * @constructor
+ * @param {function(function(*=), function (*=))} resolver
+ */
+function Promise(resolver) {
```

Comment:
Yup, [it's compliant](https://github.com/jridgewell/PJs/blob/master/test/promises-spec.js).